src/main/java/com/users/service/UserService.java

```diff
@@ -36,6 +36,15 @@ public UserDto create(UserDto user) {
         if (userRepository.existsByPrimaryEmail(user.getPrimaryEmail())) {
             throw new EntityAlreadyExistsException(String.format("User with email %s already exists", user.getPrimaryEmail()));
         }
+        for(String userSecondaryEmail : user.getSecondaryEmails()){
+            if(userRepository.existsByPrimaryEmail(userSecondaryEmail)){
+                throw new EntityAlreadyExistsException((String.format("Secondary email %s already exists as primary", userSecondaryEmail)));
+            }
+            if(userSecondaryEmail.equals(user.getPrimaryEmail())){
```

Comment:
in primul rand ai acelasi mesaj de eroare pentru fiecare if, gen daca vei primi mesajul nu vezi sti daca eroare e de la faptul ca ai deja in db acel email ca primary, sau ca in request.. 2 la mana e mai indicat sa faci prima data al doilea if in loc sa faci 10 request uri in db ca sa observi dupa ca defapt era problema in request..si cred ca exista si o metoda mai simpla sa verifici daca o lista de string uri contine un string